controllers/installation_controller.go

```diff
@@ -119,29 +116,42 @@ func (r *InstallationReconciler) ReconcileInstallation(ctx context.Context, in *
 			trimmed = append(trimmed, nodeStatus)
 			continue
 		}
-		needsUpdate = true
-		rmevent := metrics.NodeRemovedEvent{ClusterID: in.Spec.ClusterID, NodeName: nodeStatus.Name}
 		if in.Spec.AirGap {
 			continue
 		}
+		rmevent := metrics.NodeRemovedEvent{
+			ClusterID: in.Spec.ClusterID,
+			NodeName:  nodeStatus.Name,
+		}
 		if err := metrics.NotifyNodeRemoved(ctx, in.Spec.MetricsBaseURL, rmevent); err != nil {
 			return fmt.Errorf("failed to notify node removed: %w", err)
 		}
 	}
-	log := ctrl.LoggerFrom(ctx)
 	sort.SliceStable(trimmed, func(i, j int) bool { return trimmed[i].Name < trimmed[j].Name })
 	in.Status.NodesStatus = trimmed
-	if !needsUpdate {
-		log.Info("No node changes detected")
-		return nil
-	}
-	log.Info("Node changes detected, updating installation status")
 	if err := r.Status().Update(ctx, in); err != nil {
 		return fmt.Errorf("failed to update installation status: %w", err)
 	}
 	return nil
 }
 
+// copyLastNodeStatus makes sure that we copied the last populated node status from the
+// previous installation objects. This is needed because we don't want to lose the last
+// node status when we create a new installation object. items is expected to be sorted
+// in descending order by creation timestamp.
```

Comment:
should we add a check to the beginning of this to ensure that the items are sorted?

Reply:
or just do the sorting here?